File: tests/expected/str_enum.nim
import tables

type Colors = enum
  RED = "red",
  GREEN = "green",
  BLUE = "blue",


proc show() =
  let color_map = {Colors.RED: "1", Colors.GREEN: "2",
      Colors.BLUE: "3"}.newTable
  let a = Colors.GREEN
  if a == Colors.GREEN:
    echo "green"
  else:
    echo "Not green"
  echo len(color_map)

proc main() =
  show()

main()

File: tests/expected/str_enum.go
package main

import (
	"fmt"
)

type Colors string

const (
	RED   Colors = "red"
	GREEN        = "green"
	BLUE         = "blue"
)

func Show() {
	color_map := map[Colors]string{RED: "1", GREEN: "2", BLUE: "3"}
	var a Colors = GREEN
	if a == GREEN {
		fmt.Printf("%v\n", "green")
	} else {
		fmt.Printf("%v\n", "Not green")
	}
	fmt.Printf("%v\n", len(color_map))
}

func main() {
	Show()
}
